File: SDKs/googleapis.com/drive/v2/go/pkg/models/operations/drivefileswatch.go
package operations

import (
	"openapi/pkg/models/shared"
)

type DriveFilesWatchPathParams struct {
	FileID string `pathParam:"style=simple,explode=false,name=fileId"`
}

type DriveFilesWatchProjectionEnum string

const (
	DriveFilesWatchProjectionEnumBasic DriveFilesWatchProjectionEnum = "BASIC"
	DriveFilesWatchProjectionEnumFull  DriveFilesWatchProjectionEnum = "FULL"
)

type DriveFilesWatchQueryParams struct {
	AcknowledgeAbuse          *bool                          `queryParam:"style=form,explode=true,name=acknowledgeAbuse"`
	Alt                       *shared.AltEnum                `queryParam:"style=form,explode=true,name=alt"`
	Fields                    *string                        `queryParam:"style=form,explode=true,name=fields"`
	IncludeLabels             *string                        `queryParam:"style=form,explode=true,name=includeLabels"`
	IncludePermissionsForView *string                        `queryParam:"style=form,explode=true,name=includePermissionsForView"`
	Key                       *string                        `queryParam:"style=form,explode=true,name=key"`
	OauthToken                *string                        `queryParam:"style=form,explode=true,name=oauth_token"`
	PrettyPrint               *bool                          `queryParam:"style=form,explode=true,name=prettyPrint"`
	Projection                *DriveFilesWatchProjectionEnum `queryParam:"style=form,explode=true,name=projection"`
	QuotaUser                 *string                        `queryParam:"style=form,explode=true,name=quotaUser"`
	RevisionID                *string                        `queryParam:"style=form,explode=true,name=revisionId"`
	SupportsAllDrives         *bool                          `queryParam:"style=form,explode=true,name=supportsAllDrives"`
	SupportsTeamDrives        *bool                          `queryParam:"style=form,explode=true,name=supportsTeamDrives"`
	UpdateViewedDate          *bool                          `queryParam:"style=form,explode=true,name=updateViewedDate"`
	UserIP                    *string                        `queryParam:"style=form,explode=true,name=userIp"`
}

type DriveFilesWatchSecurityOption1 struct {
	Oauth2  shared.SchemeOauth2  `security:"scheme,type=oauth2"`
	Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
}

type DriveFilesWatchSecurityOption2 struct {
	Oauth2  shared.SchemeOauth2  `security:"scheme,type=oauth2"`
	Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
}

type DriveFilesWatchSecurityOption3 struct {
	Oauth2  shared.SchemeOauth2  `security:"scheme,type=oauth2"`
	Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
}

type DriveFilesWatchSecurityOption4 struct {
	Oauth2  shared.SchemeOauth2  `security:"scheme,type=oauth2"`
	Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
}

type DriveFilesWatchSecurityOption5 struct {
	Oauth2  shared.SchemeOauth2  `security:"scheme,type=oauth2"`
	Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
}

type DriveFilesWatchSecurityOption6 struct {
	Oauth2  shared.SchemeOauth2  `security:"scheme,type=oauth2"`
	Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
}

type DriveFilesWatchSecurityOption7 struct {
	Oauth2  shared.SchemeOauth2  `security:"scheme,type=oauth2"`
	Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
}

type DriveFilesWatchSecurity struct {
	Option1 *DriveFilesWatchSecurityOption1 `security:"option"`
	Option2 *DriveFilesWatchSecurityOption2 `security:"option"`
	Option3 *DriveFilesWatchSecurityOption3 `security:"option"`
	Option4 *DriveFilesWatchSecurityOption4 `security:"option"`
	Option5 *DriveFilesWatchSecurityOption5 `security:"option"`
	Option6 *DriveFilesWatchSecurityOption6 `security:"option"`
	Option7 *DriveFilesWatchSecurityOption7 `security:"option"`
}

type DriveFilesWatchRequest struct {
	PathParams  DriveFilesWatchPathParams
	QueryParams DriveFilesWatchQueryParams
	Request     *shared.Channel `request:"mediaType=application/json"`
	Security    DriveFilesWatchSecurity
}

type DriveFilesWatchResponse struct {
	Channel     *shared.Channel
	ContentType string
	StatusCode  int64
}

File: SDKs/googleapis.com/drive/v3/typescript/src/sdk/models/operations/drivefileswatch.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class DriveFilesWatchPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=fileId" })
  fileId: string;
}


export class DriveFilesWatchQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=acknowledgeAbuse" })
  acknowledgeAbuse?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=alt" })
  alt?: shared.AltEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=fields" })
  fields?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=includeLabels" })
  includeLabels?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=includePermissionsForView" })
  includePermissionsForView?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=key" })
  key?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=oauth_token" })
  oauthToken?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=prettyPrint" })
  prettyPrint?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=quotaUser" })
  quotaUser?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=supportsAllDrives" })
  supportsAllDrives?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=supportsTeamDrives" })
  supportsTeamDrives?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=userIp" })
  userIp?: string;
}


export class DriveFilesWatchSecurityOption1 extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class DriveFilesWatchSecurityOption2 extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class DriveFilesWatchSecurityOption3 extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class DriveFilesWatchSecurityOption4 extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class DriveFilesWatchSecurityOption5 extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class DriveFilesWatchSecurityOption6 extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class DriveFilesWatchSecurityOption7 extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class DriveFilesWatchSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, option=true" })
  option1?: DriveFilesWatchSecurityOption1;

  @SpeakeasyMetadata({ data: "security, option=true" })
  option2?: DriveFilesWatchSecurityOption2;

  @SpeakeasyMetadata({ data: "security, option=true" })
  option3?: DriveFilesWatchSecurityOption3;

  @SpeakeasyMetadata({ data: "security, option=true" })
  option4?: DriveFilesWatchSecurityOption4;

  @SpeakeasyMetadata({ data: "security, option=true" })
  option5?: DriveFilesWatchSecurityOption5;

  @SpeakeasyMetadata({ data: "security, option=true" })
  option6?: DriveFilesWatchSecurityOption6;

  @SpeakeasyMetadata({ data: "security, option=true" })
  option7?: DriveFilesWatchSecurityOption7;
}


export class DriveFilesWatchRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: DriveFilesWatchPathParams;

  @SpeakeasyMetadata()
  queryParams: DriveFilesWatchQueryParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: shared.Channel;

  @SpeakeasyMetadata()
  security: DriveFilesWatchSecurity;
}


export class DriveFilesWatchResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  channel?: shared.Channel;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
